🍒 [lldb] make lit use the same PYTHONHOME for building and testing (#143183)

charles-zablit opened the pull request:
- https://github.com/llvm/llvm-project/pull/143183
- https://github.com/llvm/llvm-project/pull/143738

charles-zablit committed:
[lldb] make lit use the same PYTHONHOME for building and testing (#143183)

When testing LLDB, we want to make sure to use the same Python as the
one we used to build it.

This patch used the CMake variable `Python3_ROOT_DIR` to set the
`PYTHONHOME` env variable in LLDB lit tests, in order to ensure of this.

Please see https://github.com/swiftlang/swift/pull/82063 for the
original issue.

adrian-prantl commented on compiler-rt/test/lit.common.configured.in:
```diff
@@ -25,6 +25,7 @@ set_default("gold_executable", "@GOLD_EXECUTABLE@")
 set_default("clang", "@COMPILER_RT_RESOLVED_TEST_COMPILER@")
 set_default("compiler_id", "@COMPILER_RT_TEST_COMPILER_ID@")
 set_default("python_executable", "@Python3_EXECUTABLE@")
+set_default("python_root_dir", "@Python3_ROOT_DIR@")
```
Why is this change in compiler-rt necessary?

charles-zablit replied:
It is not. I just checked by rerunning the tests without that setting. I thought this would set the default value for `python_root_dir` globally to all lit tests but this is not the case.

I will open a PR on upstream to remove it and cherry-pick it back to this PR.

charles-zablit replied:
PR opened: https://github.com/llvm/llvm-project/pull/143738

charles-zablit replied:
Fixed 👍 
Should I squash the commits?

adrian-prantl replied:
Not if they are cherry-picks.

charles-zablit committed:
[compiler-rt] remove unused default in compiler-rt lit tests (#143738)

In https://github.com/llvm/llvm-project/pull/143183 was mistakenly added
a default value to `python_root_dir` in lit tests of compiler-rt.

This is unused by the lit tests of compiler-rt, as it was meant to be
used by `lldb`.

This patch removes this change.

charles-zablit commented:
@swift-ci please test

charles-zablit commented:
@swift-ci please test macOS
